projects/angular-auth-oidc-client/src/lib/validation/token-validation.service.ts

```diff
@@ -261,6 +262,28 @@ export class TokenValidationService {
         return true;
     }
 
+    validateIdTokenAzpExistsIfMoreThanOneAud(dataIdToken: any, aud: any): boolean {
+        if (dataIdToken.aud instanceof Array && dataIdToken.aud.length > 1) {
+            if (!dataIdToken.azp) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+    // If an azp (authorized party) Claim is present, the Client SHOULD verify that its client_id is the Claim Value.
+    validateIdTokenAzpValid(dataIdToken: any, clientId: string): boolean {
+        if (!dataIdToken.azp) {
```

Comment:
add a questionmark --> ` if (!dataIdToken?.azp) {` checks for null and the `azp` property

Reply:
fixed